### PR TITLE
Finish relay

### PR DESCRIPTION
**Pull request by PatStiles**
Reopening this pr for EPF project

### CODE REVIEW

File: mev-relay-rs/src/relay.rs [hunk 1]
```diff
@@ -104,24 +124,24 @@ pub struct Inner {
...
         let validator_registry = ValidatorRegistry::new(beacon_node);
+
         let inner = Inner {
             secret_key,
             public_key,
             genesis_validators_root,
             builder: NullBuilder,
             validator_registry,
             context,
+            //TODO: Can we rely on the future to poll and update this upon initialization instead of initializing it?
```
Comment (ralexstokes) [inline mev-relay-rs/src/relay.rs:144]: no

it just allocates the default state here, I don't see a reason to move this somewhere else

**Comment by ralexstokes** [comment]
@PatStiles thanks for opening this!

I intend to do some heavy refactoring of the relay code soon, can you just work on filling out the verification functions (which it looks like you have done most of that here) and leaving other parts alone

in particular we will need to set up reth simulation infra and its a bit tricky as you can see in #112

**Comment by ralexstokes** [comment]
to add some more color here, many of the remaining validations require integration with a CL or EL

doing this integration will likely substantially change the layout of the code here

I plan to do this soon but haven't had time to yet

I'd even suggest holding off on this until it is clearer what is useful work here

**Comment by PatStiles** [comment]
@ralexstokes Thanks for adding color to this! Makes a lot of sense! Will circle back around once the refactors are complete.

**Comment by PatStiles** [comment]
Closing this in favor of #132

**Pull request closed without merging**